pkg/controller/bastion/actuator.go

```diff
@@ -0,0 +1,219 @@
+}
+
+func createBastionInstance(openstackClientFactory openstackclient.Factory, parameters servers.CreateOpts) (*servers.Server, error) {
+	computerclient, err := openstackClientFactory.Compute()
+	if err != nil {
+		return nil, err
+	}
+	return computerclient.CreateServer(parameters)
+}
+
+func deleteBastionInstance(openstackClientFactory openstackclient.Factory, id string) error {
+	computerclient, err := openstackClientFactory.Compute()
+	if err != nil {
+		return err
+	}
+	return computerclient.DeleteServer(id)
+}
+
+// GetIPs return privateip, publicip
+func GetIPs(s *servers.Server, opt *Options) (string, string, error) {
+	var privateIP, publicIp string
+
+	type InstanceNic struct {
+		MacAddr string `json:"OS-EXT-IPS-MAC:mac_addr"`
+		Version int    `json:"version"`
+		Addr    string `json:"addr"`
+		Type    string `json:"OS-EXT-IPS:type"`
+	}
+
+	instanceNic := []InstanceNic{}
+
+	if len(s.Addresses) == 0 {
+		return "", "", fmt.Errorf("NIC not ready yet")
+	}
+
+	bytes, err := json.Marshal(s.Addresses[opt.ShootName])
+	if err != nil {
+		return "", "", err
+	}
+	err = json.Unmarshal(bytes, &instanceNic)
+	if err != nil {
+		return "", "", err
+	}
+
+	for i, v := range instanceNic {
+		if v.Type == "fixed" {
+			privateIP = instanceNic[i].Addr
+		} else {
+			publicIp = instanceNic[i].Addr
+		}
+	}
+
+	return privateIP, publicIp, nil
+
+}
+
+func createFloatingIP(openstackClientFactory openstackclient.Factory, parameters floatingips.CreateOpts) (*floatingips.FloatingIP, error) {
+	client, err := openstackClientFactory.Networking()
+	if err != nil {
+		return nil, err
+	}
+	return client.CreateFloatingIP(parameters)
+}
+
+func deleteFloatingIP(openstackClientFactory openstackclient.Factory, id string) error {
+	client, err := openstackClientFactory.Networking()
+	if err != nil {
+		return err
+	}
+	return client.DeleteFloatingIP(id)
+}
+
+func associateFIPWithInstance(openstackClientFactory openstackclient.Factory, id string, parameter computerfip.AssociateOpts) error {
+	client, err := openstackClientFactory.Compute()
+	if err != nil {
+		return err
+	}
+	return client.AssociateFIPWithInstance(id, parameter)
+}
+
+func findFloatingIDbyInstnaceID(openstackClientFactory openstackclient.Factory, id string) (string, error) {
+	client, err := openstackClientFactory.Compute()
+	if err != nil {
+		return "", err
+	}
+	return client.FindFloatingIDbyInstnaceID(id)
+}
+
+func getExternalNetworkInfoByName(openstackClientFactory openstackclient.Factory, name string) ([]networks.Network, error) {
+	client, err := openstackClientFactory.Networking()
+	if err != nil {
+		return nil, err
+	}
+	return client.GetExternalNetworkInfoByName(name)
+
+}
+
+func getFipbyName(openstackClientFactory openstackclient.Factory, name string) ([]floatingips.FloatingIP, error) {
+	client, err := openstackClientFactory.Networking()
+	if err != nil {
+		if ignoreNotFoundError(err) == nil {
+			return nil, nil
+		}
+		return nil, err
+	}
+	return client.GetFipbyName(name)
```

Comment:
committed in 4d44ae7

Reply:
This is not a problem of using the helper function from the client package but I want to understand if there is a case  where the client construction would fail due to a NotFoundError and why ignoring it would help.

Reply:
sorry the wrong hash pasted here, it should be 508e7fd952e507cfa1ddbb64e572f19473edef63 and the NotfoundError removed